recipes/aws-sdk-cpp/all/conanfile.py

```diff
@@ -312,23 +312,42 @@ class AwsSdkCppConan(ConanFile):
     def _source_subfolder(self):
         return "source_subfolder"
 
+    @property
+    def _use_aws_crt_cpp(self):
+        return tools.Version(self.version) >= "1.9"
+
     def config_options(self):
         if self.settings.os == "Windows":
             del self.options.fPIC
 
     def configure(self):
         if self.options.shared:
             del self.options.fPIC
-            if (self.settings.compiler == "gcc"
-                    and tools.Version(self.settings.compiler.version) < "6.0"):
-                raise ConanInvalidConfiguration("""Doesn't support gcc5 / shared.
+
+    def validate(self):
+        if (self.options.shared
+            and self.settings.compiler == "gcc"
+            and tools.Version(self.settings.compiler.version) < "6.0"):
+            raise ConanInvalidConfiguration("""Doesn't support gcc5 / shared.
                 See https://github.com/conan-io/conan-center-index/pull/4401#issuecomment-802631744""")
+        if (not self._use_aws_crt_cpp
+            and self.settings.os == "Macos"
+            and self.settings.arch == "armv8"):
+            raise ConanInvalidConfiguration("""This version doesn't support arm8
+                See https://github.com/aws/aws-sdk-cpp/issues/1542""")
 
     def requirements(self):
-        self.requires("aws-c-event-stream/0.1.5")
+        self.requires("aws-c-common/0.6.9")
+        if self._use_aws_crt_cpp:
+            self.requires("aws-c-cal/0.5.12")
+            self.requires("aws-c-io/0.10.5")
+            self.requires("aws-crt-cpp/0.14.3")
+        else:
+            self.requires("aws-c-event-stream/0.1.5")
```

Comment:
I wonder what is the best option here: requiring different version for `aws-c-common` depending on the `_use_aws_crt_cpp` property or bump `aws-c-common` in the `aws-c-event-stream` recipe.